addons/website_slides/models/slide_channel.py

```diff
@@ -331,16 +331,18 @@ def _compute_has_requested_access(self):
 
     @api.depends('channel_partner_ids.partner_id')
     @api.depends_context('uid')
-    @api.model
     def _compute_is_member(self):
+        if self.env.user is self.env.ref('base.public_user'):
+            self.is_member = False
+            return
```

Comment:
Should be able to use `self.env.user._is_public` ?

Reply:
Actually, no (this is what broke the survey test in previous build).
Someone can be invited with a token to access a course, in that case they'll be _is_public() but linked to a partner and can be "student" of the course , so we have to continue here.
Unless I'm wrong, a simple anonymous browsing the website *is* `base.public_user`? That is why I added this section here.

Reply:
From what I understand, public users' partner_id field will be "base.public_partner", so I still don't understand how this makes any difference?

I don't think a user returning "is_public true" can be "linked to a partner that is member of the course".

Reply:
I still don't think that public users can be members of a slide channel 🤔 

So I would say the "test_course_certification_failure" is wrong and needs to use a portal user instead.

(Shame on the guy that wrote this test, booooh! 🥸 ).